### PR TITLE
Improve sub and unsub errors (RIPD-702)

### PR DESCRIPTION
**Pull request by miguelportilla**
Reviewers: @ximinez @seelabs

### CODE REVIEW

**Comment by seelabs** [comment]
:+1: Very nice clean ups in addition to the new error checks

File: src/ripple/rpc/handlers/Unsubscribe.cpp [hunk 1]
```diff
@@ -37,98 +37,91 @@ Json::Value doUnsubscribe (RPC::Context& context)
...
+        if (! context.params[jss::streams].isArray ())
+            return rpcError (rpcINVALID_PARAMS);
+
         for (auto& it: context.params[jss::streams])
         {
-            if (it.isString ())
-            {
-                std::string streamName = it.asString ();
+            if (! it.isString())
+                RPC::inject_error(rpcSTREAM_MALFORMED, jvResult);
```
Comment (ximinez) [inline src/ripple/rpc/handlers/Unsubscribe.cpp]: This is injecting the error, but then going on to attempt processing of `it`, which will, of course, throw an exception when the next line calls `asString()`. I would assume you want to either `continue` or `return` after the injection.
```
            if (! it.isString())
            {
                RPC::inject_error(rpcSTREAM_MALFORMED, jvResult);
                continue;
            }
```

**Comment by ximinez** [comment]
Other than the error injection, which is a pretty big deal, these changes look great. Let me know when you've fixed that, and I'll sign off.

File: src/ripple/rpc/handlers/Unsubscribe.cpp (hunk 1 above)
Comment (miguelportilla) [inline src/ripple/rpc/handlers/Unsubscribe.cpp]: Good catch, I meant return rpcError. Same problem in the last else. Thanks!

**Comment by ximinez** [comment]
:+1: contingent on fixing the typo above. Feel free to mark this as "Passed" once you've fixed that.

**Pull request closed without merging**